### PR TITLE
UHM-4975, do not migrate obs from test patients

### PR DESCRIPTION
**Pull request by cioan**
I think these are so far all places that we needed to filter out test patient data.

### CODE REVIEW

File: etl/src/main/java/org/pih/hivmigration/etl/sql/ExamLabResultsMigrator.groovy [hunk 1]
```diff
@@ -19,8 +19,9 @@ class ExamLabResultsMigrator extends ObsMigrator {
             ) VALUES (?, ?, ?, ?)
         ''',
         '''
-            SELECT encounter_id, lab_test, result, test_date 
-            FROM HIV_EXAM_LAB_RESULTS
+            SELECT r.encounter_id, r.lab_test, r.result, r.test_date 
+            FROM HIV_EXAM_LAB_RESULTS r, hiv_encounters e, hiv_demographics_real d 
+            WHERE r.encounter_id=e.encounter_id and e.patient_id = d.patient_id
```
Comment (brandones) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/ExamLabResultsMigrator.groovy:24]:
```suggestion
            WHERE r.encounter_id = e.encounter_id and e.patient_id = d.patient_id
```

File: etl/src/main/java/org/pih/hivmigration/etl/sql/TreatmentObsMigrator.groovy [hunk 2]
```diff
@@ -23,8 +22,9 @@ class TreatmentObsMigrator extends ObsMigrator {
             INSERT INTO hivmigration_ordered_other (source_encounter_id, ordered, comments)
             VALUES (?, ?, ?)
         ''', '''
-            SELECT encounter_id, ordered, comments
-            FROM hiv_ordered_other
+            SELECT o.encounter_id, o.ordered, o.comments
+            FROM hiv_ordered_other o, hiv_encounters e, hiv_demographics_real d 
+            WHERE o.encounter_id=e.encounter_id and e.patient_id=d.patient_id
```
Comment (brandones) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/TreatmentObsMigrator.groovy:27]:
```suggestion
            WHERE o.encounter_id = e.encounter_id and e.patient_id = d.patient_id
```

**Review by brandones — APPROVED** [review]

**Review by mogoodrich — APPROVED** [review]
LGTM!  

Didn't do any checking to see if there were potentially any places this join was missed though... I figure we can just continue to test as we go forward.

Also, defer to you guys about coordinating moving forward with Mike's idea of creating a staging obs table in OpenMRS and doing the filtering in that initial import and then referencing that staging table in all these queries.

**Comment by cioan** [comment]
Thanks @mogoodrich and @brandones !

**Pull request merged**